ui/src/notification_bar.tsx

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020 H2O.ai, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as Fluent from '@fluentui/react'
+import { B, Box, box, S, U } from 'h2o-wave'
+import React from 'react'
+import { Component } from './form'
+import { grid } from './layout'
+import { MessageBar } from "./message_bar"
+import { bond } from './ui'
+
+/**
+ * Create a notification bar.
+ *
+ * A notification bar is an area at the edge of a primary view that displays relevant status information.
+ * You can use a notification bar to tell the user about a result of an action, e.g. "Data has been successfully saved".
+ */
+export interface NotificationBar {
+  /** The text displayed on the notification bar. */
+  text: S
+  /** The icon and color of the notification bar. Defaults to 'info'. */
+  type?: 'info' | 'error' | 'warning' | 'success' | 'danger' | 'blocked'
+  /** When should the notification bar disappear in seconds. Defaults to 5. */
+  timeout?: U
+  /** Specify one or more action buttons. */
+  buttons?: Component[]
+  /** Specify the location of notification. Defaults to 'top-right'. */
+  position?: 'top-right' | 'bottom-right' | 'bottom-center' | 'bottom-left' | 'top-left' | 'top-center'
+  /** The events to capture on this notification bar. */
+  events?: S[]
```

Comment:
Do we need an `auto_hide` (default to true) like in Bootstrap's toast component?

Reply:
It currently works like this: auto-close after 5s (adjustable timeout) when no buttons are included (if included, we assume an action is required, thus the notification is not closed). Every notification includes an `X` button to manually dismiss the notification if needed.

This is good enough IMO, but open to suggestions.